client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx

```diff
@@ -81,213 +98,198 @@ beforeEach(() => {
+      await screen.findByText('Review & Launch')
+      expect(container).toMatchSnapshot()
+    })
   })
 
   it('renders despite missing jurisdictions on opportunistic contest', async () => {
-    apiMock.mockImplementation(
-      generateApiMock(
-        sampleSizeMock,
-        contestMocks.filledOpportunisticWithJurisdictionId,
-        {
-          jurisdictions: [],
-        }
-      )
-    )
     auditSettingsMock.mockReturnValue(settingsMock.full)
-    const { container } = render(
-      <Router>
-        <Review locked={false} prevStage={prevStage} refresh={jest.fn()} />
-      </Router>
-    )
-    await waitFor(() => expect(apiMock).toHaveBeenCalled())
-    expect(container).toMatchSnapshot()
+    const expectedCalls = [
+      apiCalls.getEmptyJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getOpportunisticContests,
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      const { container } = renderView()
+      await screen.findByText('Review & Launch')
+      expect(container).toMatchSnapshot()
+    })
   })
 
   it('launches the first round', async () => {
-    apiMock.mockImplementation(
-      generateApiMock(
-        sampleSizeMock,
-        contestMocks.filledTargetedWithJurisdictionId,
-        { jurisdictions: jurisdictionMocks.allManifests }
-      )
-    )
-    render(
-      <Router>
-        <Review locked={false} prevStage={prevStage} refresh={jest.fn()} />
-      </Router>
-    )
-    const launchButton = await screen.findByText('Launch Audit')
-    fireEvent.click(launchButton, { bubbles: true })
-    await screen.findByText('Are you sure you want to launch the audit?')
-    const confirmLaunchButton = screen.getAllByText('Launch Audit')[1]
-    fireEvent.click(confirmLaunchButton, { bubbles: true })
-    await waitFor(() => {
-      expect(apiMock).toHaveBeenCalledTimes(5)
-      expect(apiMock.mock.calls[4][1]).toMatchObject({
-        body: JSON.stringify({
-          sampleSize: 46,
-          roundNum: 1,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        method: 'POST',
-      })
+    auditSettingsMock.mockReturnValue(settingsMock.full)
+    const expectedCalls = [
+      apiCalls.getJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getTargetedContests,
+      apiCalls.getSampleSizeOptions,
+      apiCalls.postRound({ 'contest-id': 46 }),
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      renderView()
+      await screen.findByText('Review & Launch')
+      const launchButton = screen.getByText('Launch Audit')
+      userEvent.click(launchButton)
+      await screen.findByText('Are you sure you want to launch the audit?')
+      const confirmLaunchButton = screen.getAllByText('Launch Audit')[1]
+      userEvent.click(confirmLaunchButton)
+      await waitFor(() => expect(refreshMock).toHaveBeenCalled())
```

Comment:
another option would be to find something on the next screen that indicates you were successfully redirected/the audit got launched. i suggest that because it's more in line with the philosophy of testing from the user's perspective

Reply:
@jonahkagan It was either check that the mock for `refresh` was called, or that the mock for `history.push` was called. The former just seemed simpler. Both happen when it is successfully launched, though.

Reply:
oh right, forgot there were no other routes to navigate to in this test setup